WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift

```diff
@@ -193,10 +193,20 @@ open class ThemeBrowserCell: UICollectionViewCell {
         activityView.stopAnimating()
     }
 
-    fileprivate func refreshScreenshotImage(_ imageUrl: String) {
+    private func getImageUrlForWidth(imageUrl: String) -> String {
         // Themes not hosted on WP.com have an incorrect screenshotUrl and do not correctly support the w param
-        let imageUrlForWidth = imageUrl.hasPrefix("http") ? imageUrl + "?w=\(presenter!.screenshotWidth)" :
-            String(format: "http:%@", imageUrl)
+        if false == imageUrl.hasPrefix("http") {
+            return String(format: "http:%@", imageUrl)
+        }
+
+        if false == imageUrl.contains("?") {
+            return imageUrl + "?w=\(presenter!.screenshotWidth)"
+        }
+        return imageUrl + "&w=\(presenter!.screenshotWidth)"
+    }
+
+    fileprivate func refreshScreenshotImage(_ imageUrl: String) {
+        let imageUrlForWidth = self.getImageUrlForWidth( imageUrl: imageUrl )
```

Comment:
NitPick: `self` shouldn't be needed here.
```suggestion
        let imageUrlForWidth = imageUrlForWidth( imageUrl: imageUrl )
```

Reply:
Remove self and update method name.